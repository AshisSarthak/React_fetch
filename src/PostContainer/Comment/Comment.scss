.comment-author {
    display: flex;
    align-items: center;
    justify-content: space-between;
}
.comment-cont {
    padding-left: 20px;
    border-bottom: 1px solid #eee;
    padding-bottom: 10px;

    .body {
        color: #666;
    }
}