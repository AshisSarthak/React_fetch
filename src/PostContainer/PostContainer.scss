.post-container {
    width: 60%;
    max-height: 400px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: 1px 0px 2px #aaa;
    overflow: auto;
    margin: 2% 20%;
}
.post-header, .commentSection{
    width: 90%;
    margin-left: 5%;
    border-bottom: 1px solid #eee; 
}
.author {
    display: flex;
    align-items: center;

    h5 {
        margin-left: 15px;
    }
}
