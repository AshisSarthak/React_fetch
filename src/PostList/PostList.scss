.post-list {
    width: 50%;
    max-height: 400px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: 1px 0px 2px #aaa;
    overflow: auto;
    margin: 2% 25%;
    cursor: pointer;
}