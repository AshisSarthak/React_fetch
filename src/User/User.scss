.user-detail {
    width: 50%;
    max-height: 400px;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: 1px 0px 2px #aaa;
    overflow: auto;
    margin: 2% 25%;
    cursor: pointer;
}

table {
    width: 100%;
    border-collapse: collapse;
    border-color: #aaa;

    tr {
        border: 1px solid #eee;
    }

    .label {
        border-right: 1px solid #eee;
        padding: 10px;
        text-align: right;
    }
    .values {
        text-indent: 10px;
    }
}
.no-collapse{
    border-collapse: separate;
    .label, .values {
        border-bottom: 1px solid #eee;
        
    }
    tr:last-child{
        .label, .values {
            border-bottom: none;
        }
    }
}