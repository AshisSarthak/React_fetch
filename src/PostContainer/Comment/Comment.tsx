import React from 'react';
import './Comment.scss';
import { PostComment } from '../PostContainer';

export const Comment = (props:{item:  PostComment}) => {
    return (
        <div className='comment-cont'>
            <div className='comment-author'>
                <h5 className="subject">
                    {props.item.name}
                </h5>
                <div className='author-name'>
                    {props.item.email}
                </div>
            </div>
            <div className="body">
                {props.item.name}
            </div>
        </div>
    )
}