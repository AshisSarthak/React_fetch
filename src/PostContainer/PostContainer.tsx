import React from 'react';
import { PostItem } from '../PostList/PostList';
import './PostContainer.scss';
import { UserDetails } from '../User/User';
import { Comment } from './Comment/Comment';

interface PostContainerProp {
  post: PostItem;
}

export interface PostComment {
  postid: number;
  id: number;
  name: string;
  email: string;
  body: string; 
}
interface PostContainerState {
  comments?: PostComment[];
  createdBy?: UserDetails;
}

export class PostContainer extends React.Component<PostContainerProp, PostContainerState> {
  constructor(props: PostContainerProp) {
    super(props);
    this.state = {
      comments: [],
      createdBy: undefined
    }
  }

  componentDidMount(){
    fetch(`https://jsonplaceholder.typicode.com/posts/${this.props.post.id}/comments`)
    .then(response => response.json())
    .then(json => this.setState({
      comments: json,
    }));
    fetch(`https://jsonplaceholder.typicode.com/users/${this.props.post.userId}`)
    .then(response => response.json())
    .then(json => this.setState({
      createdBy: json,
    }));
  }

  render(){
    return (
      <div className="post-container">
        <div className="post-header">
          <h4>{this.props.post.title}</h4>
          <div className="author">
            <sub>Created By </sub> <h5>{this.state.createdBy && this.state.createdBy.name}</h5> 
          </div>
        </div>
        <div className="commentSection">
          {
            this.state.comments && this.state.comments.map((val, index) => {
              return <Comment item={val} key={index} />
            })
          }
        </div>

      </div>
   );  
  }
}

export default PostContainer;
