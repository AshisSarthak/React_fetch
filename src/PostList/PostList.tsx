import React, { ReactNode } from 'react';
import PostItemEl from './PostItem/PostItem';
import "./PostList.scss";
import { UserDetails } from '../User/User';

export interface PostItem {
  userId: number;
  id: number;
  title: string;
  body: string;
}

interface PostListState {
  postList: PostItem[];
}
interface PostListProp {
  selectUser: (user:UserDetails) => void;
  selectPost: (post:PostItem) => void;
}
export class PostList extends React.Component<PostListProp, PostListState> {

  constructor(props:PostListProp, state:PostListState){
    super(props, state);
    this.state = {
      postList:[]
    };
  }

  componentDidMount(){
    fetch('https://jsonplaceholder.typicode.com/posts')
  .then(response => response.json())
  .then(json => this.setState({
    postList: json
  }));
  }

  generatePost():ReactNode {
    return this.state.postList.map((item, index)=>{
      return(
        <PostItemEl
          key={index}
          item={item}
          selectPost={this.props.selectPost}
          selectUser={this.props.selectUser}
        />
      )
    })
  }

  render() {
    return (
      <section className="post-list">
        {
          this.generatePost()
        }
      </section>
    );  
  }
}

export default PostList;
