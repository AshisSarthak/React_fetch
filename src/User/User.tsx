import React from 'react';
import './User.scss';

export interface UserDetails {
  id: number;
  name: string;
  username: string;
  email: string;
  address: {
    street: string;
    suite: string;
    city: string;
    zipcode: string;
    geo: {
      lat: string;
      lng: string
    }
  };
  phone: string;
  website: string;
  company: {
    name: string;
    catchPhrase: string;
    bs: string;
  }
}

function User(props:{user: UserDetails}) {

  return (
    <section className="user-detail">
      <table>
        <tbody>
          <tr>
            <td className='label user'>User Name</td>
            <td className='values'>{props.user.username}</td>
          </tr>
          <tr>
            <td className='label name'>Name</td>
            <td className='values'>{props.user.name}</td>
          </tr>
          <tr>
            <td className='label email'>Email</td>
            <td className='values'>{props.user.email}</td>
          </tr>
          <tr>
            <td className='label site'>Website</td>
            <td className='values'>{props.user.website}</td>
          </tr>
          <tr>
            <td className='label company'>Company</td>
            <td className='values'>
              <table className="no-collapse">
                <tbody>
                  <tr className="">
                    <td className='label'>Name</td>
                    <td className='values'>{props.user.company.name}</td>
                  </tr>
                  <tr className="">
                    <td className='label'>Catch Phrase</td>
                    <td className='values'>{props.user.company.catchPhrase}</td>
                    </tr>
                  <tr className="">
                    <td className='label'>BS</td>
                    <td className='values'>{props.user.company.bs}</td>
                  </tr>
                </tbody>
              </table>
              
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  );
}

export default User;
