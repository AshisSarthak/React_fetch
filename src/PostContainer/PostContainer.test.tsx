import React from "react";
import { shallow, mount } from "enzyme";
import { PostContainer } from "./PostContainer";
import { PostItem } from "../PostList/PostList";
import fetchMock from "jest-fetch-mock";
import { demoComment } from "./Comment/Comment.test";
import { UserDetails } from "../User/User";

const demoPost: PostItem = {
  body: "demo",
  id: 1,
  title: "demo",
  userId: 11,
};

const demoUser: UserDetails = {
  id: 1,
  name: "demo",
  username: "demo",
  email: "demo",
  address: {
    street: "demo",
    suite: "demo",
    city: "demo",
    zipcode: "demo",
    geo: {
      lat: "demo",
      lng: "demo",
    },
  },
  phone: "demo",
  website: "demo",
  company: {
    name: "demo",
    catchPhrase: "demo",
    bs: "demo",
  },
};

describe("PostContainer Componnet", () => {
  afterEach(() => {
    fetchMock.resetMocks();
  });

  it("default render", () => {
    const wrapper = shallow(<PostContainer post={demoPost} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("loads certain post", () => {
    const comments = [demoComment];
    fetchMock.mockOnce(JSON.stringify(comments));
    fetchMock.mockOnce(JSON.stringify(demoUser));
    const wrapper = mount(<PostContainer post={demoPost} />);
    expect(wrapper).toMatchSnapshot();
  });
});
