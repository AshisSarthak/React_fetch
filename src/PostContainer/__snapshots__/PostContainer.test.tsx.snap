// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Componnet default render 1`] = `
<div
  className="comment-cont"
>
  <div
    className="comment-author"
  >
    <h5
      className="subject"
    >
      demo
    </h5>
    <div
      className="author-name"
    >
      abc@demo.com
    </div>
  </div>
  <div
    className="body"
  >
    demo
  </div>
</div>
`;

exports[`PostContainer Componnet default render 1`] = `
<div
  className="post-container"
>
  <div
    className="post-header"
  >
    <h4>
      demo
    </h4>
    <div
      className="author"
    >
      <sub>
        Created By 
      </sub>
       
      <h5 />
    </div>
  </div>
  <div
    className="commentSection"
  />
</div>
`;

exports[`PostContainer Componnet loads certain post 1`] = `
<PostContainer
  post={
    Object {
      "body": "demo",
      "id": 1,
      "title": "demo",
      "userId": 11,
    }
  }
>
  <div
    className="post-container"
  >
    <div
      className="post-header"
    >
      <h4>
        demo
      </h4>
      <div
        className="author"
      >
        <sub>
          Created By 
        </sub>
         
        <h5 />
      </div>
    </div>
    <div
      className="commentSection"
    />
  </div>
</PostContainer>
`;
